react/features/filmstrip/subscriber.web.js

```diff
@@ -134,3 +140,76 @@ StateListenerRegistry.register(
     /* listener */ (widthBelowThreshold, store) => {
         store.dispatch(setOverflowDrawer(widthBelowThreshold));
     });
+
+/**
+ * Gracefully hide/show the filmstrip when going past threshold.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/base/responsive-ui'].clientWidth < ASPECT_RATIO_BREAKPOINT,
+    /* listener */ (widthBelowThreshold, store) => {
+        store.dispatch(setFilmstripVisible(!widthBelowThreshold));
+    });
+
+/**
+ * Listens for changes in the screen size to recompute
+ * the dimensions of the tile view grid and the tiles for responsiveness.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => {
+        const { clientWidth } = state['features/base/responsive-ui'];
+
+        // Forcing the recomputation of tiles when screen switches in or out of the (1000, 500] interval.
+        return clientWidth < TWO_COLUMN_BREAKPOINT && clientWidth >= ASPECT_RATIO_BREAKPOINT;
+    },
+    /* listener */ (_, store) => {
+        const state = store.getState();
+
+        if (shouldDisplayTileView(state)) {
+            const gridDimensions = getTileViewGridDimensions(state);
+            const { clientHeight, clientWidth } = state['features/base/responsive-ui'];
+            const { isOpen } = state['features/chat'];
+
+            store.dispatch(
+                setTileViewDimensions(
+                    gridDimensions,
+                    {
+                        clientHeight,
+                        clientWidth
+                    },
+                    isOpen
+                )
+            );
+        }
+    });
+
+/**
+ * Listens for changes in the screen size to recompute
+ * the dimensions of the tile view grid and the tiles for responsiveness.
+ */
+StateListenerRegistry.register(
```

Comment:
Tried to merge this listener with the one above but for some reason the interval-based conditions didn't seem to click. Any tips would be greatly appreciated 👍

Reply:
I think you can write them in a single listener and return a Symbol for each case, so the listener runs on those cases since they'd compare equal:

```
const stuff = {
  a: Symbol('a'),
  b: Symbol('b')
  c: Symbol('c')
};

if (clientWidth < TWO_COLUMN_BREAKPOINT && clientWidth >= ASPECT_RATIO_BREAKPOINT) {
    return stuff.a;
} else if (clientWidth < ASPECT_RATIO_BREAKPOINT && clientWidth >= SINGLE_COLUMN_BREAKPOINT) {
    return stuff.b;
} else {
    return stuff.c;
}

```